internal/server/v1/iam/service.go

```diff
@@ -0,0 +1,89 @@
+package iam
+
+import (
+	"context"
+
+	shieldv1beta1rpc "buf.build/gen/go/gotocompany/proton/grpc/go/gotocompany/shield/v1beta1/shieldv1beta1grpc"
+	shieldv1beta1 "buf.build/gen/go/gotocompany/proton/protocolbuffers/go/gotocompany/shield/v1beta1"
+	"google.golang.org/protobuf/types/known/structpb"
+
+	"github.com/goto/dex/warden"
+)
+
+//go:generate mockery --with-expecter --keeptree --case snake --name WardenClient
+
+type Service struct {
+	shieldClient shieldv1beta1rpc.ShieldServiceClient
+	wardenClient WardenClient
+}
+
+type WardenClient interface {
+	ListUserTeams(ctx context.Context, req warden.TeamListRequest) ([]warden.Team, error)
+	TeamByUUID(ctx context.Context, req warden.TeamByUUIDRequest) (*warden.Team, error)
+}
+
+func NewService(shieldClient shieldv1beta1rpc.ShieldServiceClient, wardenClient WardenClient) *Service {
+	return &Service{
+		shieldClient: shieldClient,
+		wardenClient: wardenClient,
+	}
+}
+
+func (svc *Service) TeamList(ctx context.Context, userEmail string) ([]warden.Team, error) {
+	teams, err := svc.wardenClient.ListUserTeams(ctx, warden.TeamListRequest{
+		Email: userEmail,
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	return teams, nil
+}
+
+func (svc *Service) UpdateGroupMetadata(ctx context.Context, groupID, wardenTeamID string) (map[string]any, error) {
```

Comment:
since it is not a generic method to update group metadata, let's name it to `LinkGroupToWarden`, wdyt?